src/users/schemas/user.schema.ts

```diff
@@ -1,6 +1,9 @@
 import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
 import mongoose, { Document } from 'mongoose';
 
+export type UserDocument = User & Document;
+
+@Schema({ collection: 'users' })
 @Schema()
 export class User extends Document {
   _id: mongoose.Types.ObjectId | string;
```

Comment:
You can import ObjectId from 'mongoose' instead of writing mongoose.Types.ObjectId